### ISSUE TITLE
NullPointerException with filesystem backend

### ISSUE THREAD

**Issue by kahing** [gaul/s3proxy#318]
step to reproduce:
1. change goofys's s3proxy backend to filesystem
2. make s3proxy.jar
3. ./test/run-tests.sh TestBenchCreate$

```
[s3proxy] W 11-08 02:02:19.424 S3Proxy-Jetty-126 o.g.s.o.e.j.server.HttpChannel:396 |::] /goofys-test-3q1bahr7ennp0x6c/test_dir/file3/
java.lang.NullPointerException: null
        at org.jclouds.blobstore.config.LocalBlobStore.getBlob(LocalBlobStore.java:644)
        at org.jclouds.blobstore.config.LocalBlobStore.getBlob(LocalBlobStore.java:201)
        at org.jclouds.blobstore.config.LocalBlobStore.blobMetadata(LocalBlobStore.java:736)
        at sun.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.google.inject.internal.DelegatingInvocationHandler.invoke(DelegatingInvocationHandler.java:37)
        at com.sun.proxy.$Proxy40.blobMetadata(Unknown Source)
        at org.gaul.s3proxy.S3ProxyHandler.handleBlobMetadata(S3ProxyHandler.java:1572)
        at org.gaul.s3proxy.S3ProxyHandler.doHandle(S3ProxyHandler.java:688)
        at org.gaul.s3proxy.S3ProxyHandlerJetty.handle(S3ProxyHandlerJetty.java:76)
```

**Comment by kahing** [comment]
[trace.log](https://github.com/gaul/s3proxy/files/3822718/trace.log)

**Comment by gaul** [comment]
Which version of S3Proxy?

**Comment by gaul** [comment]
Please test this by compiling S3Proxy using jclouds 2.2.1-SNAPSHOT.  Note that you will have to wait an hour or more for the snapshots to update.

**Comment by gaul** [comment]
Released 2.2.1 and opened kahing/goofys#518.
